File: src/uri/uri1845.py
import re

def setup():
    global low
    global up

    low = set(['j', 'p', 's', 'x', 'b', 'v', 'z'])
    up = set()
    for it in low:
        up.add(it.upper())

def draw():
    global low
    global up

    line = raw_input()
    outlet = str()

    for ch in line:
        if ch in low:
            outlet += 'f'
        elif ch in up:
            outlet += 'F'
        else:
            outlet += ch

    print re.sub(r"f+(F+|h+|H+)?", 'f', re.sub(r"F+(f+|h+|H+)?", 'F', outlet))

if __name__ == '__main__':
    setup()
    while True:
        try:
            draw()
        except:
            break

File: src/uri/uri1845.rb
require 'set'

# setup
low = Set.new ['j', 'p', 's', 'x']
up = Set.new
low.each { |c| up << c.capitalize }

# draw
while line = gets
    outlet = ''
    line.chomp.each_char do |ch|
        if up.include? ch
            outlet << 'F'
        elsif low.include? ch
            outlet << 'f'
        else
            outlet << ch
        end
    end
    outlet.gsub! /f+/,  'f'
    puts outlet
end
